client/reader/stream/index.jsx

```diff
@@ -60,10 +61,56 @@ function refreshCardFactory( post ) {
 
 const defaultCardFactory = config.isEnabled( 'reader/refresh/stream' ) ? refreshCardFactory : oldCardFactory;
 
+const MIN_DISTANCE_BETWEEN_RECS = 4; // page size is 7, so one in the middle of every page and one on page boundries, sometimes
+const MAX_DISTANCE_BETWEEN_RECS = 49; // page size is 7, so 7 pages
+const RECS_PER_BLOCK = 2;
+
+function getDistanceBetweenRecs() {
+	// the distance between recs changes based on how many subscriptions the user has.
+	// We cap it at MAX_DISTANCE_BETWEEN_RECS.
+	// It grows at the natural log of the number of subs, times a multiplier, offset by a constant.
+	// This lets the distance between recs grow quickly as you add subs early on, and slow down as you
+	// become a common user of the reader.
+	const totalSubs = FeedSubscriptionStore.getTotalSubscriptions();
+	return Math.min(
+		MAX_DISTANCE_BETWEEN_RECS,
+		Math.max(
+			( Math.log( totalSubs ) * Math.LOG2E * 5 ) - 6,
+			MIN_DISTANCE_BETWEEN_RECS )
+			);
+}
+
+function injectRecommendations( posts, recs = [] ) {
+	if ( ! recs || recs.length === 0 ) {
+		return posts;
+	}
+
+	const itemsBetweenRecs = getDistanceBetweenRecs();
+	// if we don't have enough posts to insert recs, don't bother
+	if ( posts.length < itemsBetweenRecs ) {
+		return posts;
+	}
+
+	const items = [];
+
+	for ( let postIndex = 0, recIndex = 0; postIndex < posts.length; postIndex += itemsBetweenRecs, recIndex += RECS_PER_BLOCK ) {
```

Comment:
This gets called when either the posts or the recs array changes.

Yeah, abusing push like this probably isn't the best method. I'm trying to do something that will allocate as few new arrays as possible, so I was trying to avoid using `concat`, which always makes a new array. 

`splice` might be the right way to go here?